src/main/java/net/sf/jabref/gui/FXAlert.java

```diff
@@ -89,6 +96,42 @@ public FXAlert(AlertType type) {
         fxDialogWindow.setOnHiding(evt -> setSwingWindowsEnabledAndFocusable(true));
 
         fxDialogWindow.setOnCloseRequest(evt -> this.close());
+
+        fxDialogWindow.getScene().setOnKeyPressed(evt -> {
+            KeyCombination combination = Globals.getKeyPrefs().getKeyCombinationFX(KeyBinding.CLOSE_DIALOG);
+            if (checkKeyCombinationEquality(combination, evt)) {
+                fxDialogWindow.close();
+            }
+        });
+    }
+
+    /**
+     * Check if the given keyCombination equals the given keyEvent
+     *
+     * @param combination as KeyCombination
+     * @param keyEvent as KeEvent
+     * @return true if matching, else false
+     */
+    private boolean checkKeyCombinationEquality(KeyCombination combination, KeyEvent keyEvent) {
+        KeyCode code = keyEvent.getCode();
+        if (code == KeyCode.UNDEFINED) {
+            return false;
+        }
+        // gather the pressed modifer keys
+        String modifiers = "";
+        if (keyEvent.isControlDown()) {
+            modifiers = "ctrl";
+        }
+        if (keyEvent.isShiftDown()) {
+            modifiers += " shift";
+        }
+        if (keyEvent.isAltDown()) {
+            modifiers += " alt";
+        }
+        modifiers = modifiers.trim();
+        String newShortcut = (modifiers.isEmpty()) ? code.toString() : modifiers + " " + code;
```

Comment:
write the above with space at the end, i.e `modifiers = "ctrl "` and `"shift "`. Then a simple `newShortcut = modifiers.trim() + code` should work.